cpp/arcticdb/version/test/test_symbol_list.cpp

```diff
@@ -153,14 +153,18 @@ TEST_P(SymbolListWithReadFailures, VersionMapSource) {
     ASSERT_THAT(symbols, UnorderedElementsAre(symbol_1, symbol_2, symbol_3));
 }
 
-INSTANTIATE_TEST_SUITE_P(, SymbolListWithReadFailures, Values(
+INSTANTIATE_TEST_SUITE_P(SymbolListWithReadFailuresSources, SymbolListWithReadFailures, Values(
         FailSimParam{}, // No failure
         FailSimParam{{FailureType::ITERATE, RAISE_ONCE}},
         FailSimParam{{FailureType::READ, RAISE_ONCE}},
         FailSimParam{{FailureType::READ, {fault(), fault(), fault(), fault(), no_op}}},
         FailSimParam{{FailureType::READ, {fault(0.4), no_op}}}, // 40% chance of exception
         FailSimParam{{FailureType::ITERATE, RAISE_ONCE}, {FailureType::READ, {no_op, fault(), no_op}}}
-));
+),
+[](auto & info) {
+      return std::string("Value")+ std::to_string(info.index);
```

Comment:
on clang this gives an invalid testname error with the name being an empty string `""`
This is probably because some compiler-dependent-magic is used to generate these names.
This was the easiest way to prevent that error

Reply:
see https://github.com/man-group/ArcticDB/actions/runs/5268343841/jobs/9524877979#step:6:7868
```
[ FATAL ] /Users/runner/micromamba/envs/arcticdb/include/gtest/internal/gtest-param-util.h:585:: Condition IsValidParamName(param_name) failed. Parameterized test name '' is invalid, in /Users/runner/work/ArcticDB/ArcticDB/cpp/arcticdb/version/test/test_symbol_list.cpp line 257
```

Reply:
(I guess you meant to reply to line 156?

I recall from the source code I read, all the test names are generated by macro expansion instead of reflection, so I don't think this is related to clang, but the version of gtest.

If you have to add a test name, can you use a shorter one, because the full name will have the `SymbolListWithReadFailures` fix class name included again.

Reply:
I think as it is, it will give enough information when a test is failing.  
Even if is a version of gtest and not the compiler, I prefer to have this merged as is, instead of debugging a strange doctest behaviour when there is a usable fix

Reply:
From my perspective, the changes that added the custom name function (which would call the tests "Value1", "Value2", etc.) are worse than the original (which would pretty-print the actual parameter values).

Looking into the failure in the output, I would presume the default `PrintToStringParamName` returned an empty string for some parameter. This is highly worrying and could indicate some kind of corruption. Please try to debug to see what happened.